src/main/java/com/redhat/rhjmc/containerjfr/commands/internal/UploadRecordingCommand.java

```diff
@@ -151,29 +159,38 @@ private ResponseMessage doPost(String targetId, String recordingName, String upl
 
     @Override
     public boolean validate(String[] args) {
-        if (args.length != 3) {
+        if (args.length != 2) {
             cw.println(
-                    "Expected three arguments: target (host:port, ip:port, or JMX service URL), recording name, and upload URL");
+                    "Expected two arguments: target (host:port, ip:port, or JMX service URL) and recording name");
             return false;
         }
 
         String targetId = args[0];
         String recordingName = args[1];
-        // String uploadUrl = args[2];
 
         boolean isValidTargetId = validateTargetId(targetId);
         if (!isValidTargetId) {
-            cw.println(String.format("%s is an invalid connection specifier", args[0]));
+            cw.println(String.format("%s is an invalid connection specifier", targetId));
         }
 
         boolean isValidRecordingName = validateRecordingName(recordingName);
         if (!isValidRecordingName) {
             cw.println(String.format("%s is an invalid recording name", recordingName));
         }
 
-        // TODO validate upload URL
+        boolean isValidDatasourceUrl =
+                new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS)
+                        .isValid(env.getEnv(GRAFANA_DATASOURCE_ENV));
+        if (!isValidDatasourceUrl) {
+            cw.println(String.format("%s is an invalid datasource URL", GRAFANA_DATASOURCE_ENV));
```

Comment:
Ah, I just noticed this on a final read-through. Should this printed error message include the (invalid) URL that was set as well, along with the name of the env var?

Reply:
Yeah, I think that would be a good idea. I'll add that now.